### PR TITLE
storybook(fxa-settings): Recreate SigninTokenCode in React

### PR DESCRIPTION
**Pull request by vpomerleau**
## Because

- We are converting content-server views to react, starting by building the UI in storybook with initial stories,  metrics and tests.

## This pull request

- Creates a new SigninTokenCode page with initial story, tests, metrics, and l10n.
- Add inputMode and tooltipPosition props to InputText component.

## Issue that this pull request solves

Closes: FXA-6421

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Previous/current page (content-server):
image

New page (React):
image

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: packages/fxa-settings/src/pages/SigninTokenCode/en.ftl [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## SigninTokenCode
+
+# String within the <span> element appears as a subheading
+signin-token-code-heading = Enter confirmation code<span> for your { -product-firefox-account }</span>
```
Comment (bcolsson) [inline packages/fxa-settings/src/pages/SigninTokenCode/en.ftl:4]: I know this string is replicating a legacy string, but the way it's presented now some languages may translate this in a different order (with the span at the beginning). I would suggest a comment with an alternative similar to the ones provided for strings in ResetPassword. Perhaps:

```suggestion
# The subheading will appear below "Enter confirmation code", so if more appropriate in a locale, the string within the <span>, "for your { -product-firefox-account }" can stand alone as "{ -product-firefox-account }"
signin-token-code-heading = Enter confirmation code<span> for your { -product-firefox-account }</span>
```

File: packages/fxa-settings/src/pages/SigninTokenCode/index.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            >
+              Confirm
+            </button>
+          </FtlMsg>
+        </form>
+        <div className="animate-delayed-fade-in opacity-0 mt-5 text-grey-500 text-xs inline-flex gap-1">
+          <FtlMsg id="signin-token-code-code-expired">
+            <p>Code expired?</p>
+          </FtlMsg>
+          {/* TODO: add button handling for resend code action
```
Comment (millsoper) [inline packages/fxa-settings/src/pages/SigninTokenCode/index.tsx:149]: What do we mean by button handling here? Like the action/cb? What are your thoughts on having this be a button with link styling?

**Review by millsoper — APPROVED** [review]
Had one question about using `<Link>` vs `<button` elements for the `Email new code` functionality, but no blockers!

File: packages/fxa-settings/src/pages/SigninTokenCode/index.tsx (hunk 2 above)
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTokenCode/index.tsx:149]: Yes, that makes sense! I dug a bit more into the resend_code mixin, and it looks like it displays the result in an alert bar and doesn't redirect. I'll update this to a button with link styling, as suggested!

File: packages/fxa-settings/src/pages/SigninTokenCode/en.ftl (hunk 1 above)
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/SigninTokenCode/en.ftl:4]: Good point. The key part of the span is that it appears on a separate line, so the following comment might provide more flexibility: 
```
# String within the <span> element appears on a separate line
# If more appropriate in a locale, the string within the <span>, "for your { -product-firefox-account }"
# can stand alone as "{ -product-firefox-account }"
```
This way, the span could be above or below "Enter confirmation code", either would be acceptable.

Comment (bcolsson) [reply packages/fxa-settings/src/pages/SigninTokenCode/en.ftl:6]: Ah great, that's perfect!

**Review by bcolsson — APPROVED** [review]

**Pull request merged**